src/dds_service.py

```diff
@@ -72,3 +74,315 @@ def builtin_observer(domain_id, dds_data, running):
                 dds_data.remove_endpoint(domain_id, sub)
 
     logging.info(f"builtin_observer({domain_id}) ... DONE")
+
+
+class dds_durability_kind(OrderedEnum):
+    DDS_DURABILITY_VOLATILE = 0
+    DDS_DURABILITY_TRANSIENT_LOCAL = 1
+    DDS_DURABILITY_TRANSIENT = 2
+    DDS_DURABILITY_PERSISTENT = 3
+
+class dds_history(OrderedEnum):
+    DDS_HISTORY_KEEP_LAST = 0
+    DDS_HISTORY_KEEP_ALL = 1
+
+class dds_ownership(OrderedEnum):
+    DDS_OWNERSHIP_SHARED = 0
+    DDS_OWNERSHIP_EXCLUSIVE = 1
+
+class dds_liveliness(OrderedEnum):
+    DDS_LIVELINESS_AUTOMATIC = 0
+    DDS_LIVELINESS_MANUAL_BY_PARTICIPANT = 1
+    DDS_LIVELINESS_MANUAL_BY_TOPIC = 2
+
+class dds_reliability(OrderedEnum):
+    DDS_RELIABILITY_BEST_EFFORT = 0
+    DDS_RELIABILITY_RELIABLE = 1
+
+class dds_destination_order(OrderedEnum):
+    DDS_DESTINATIONORDER_BY_RECEPTION_TIMESTAMP = 0
+    DDS_DESTINATIONORDER_BY_SOURCE_TIMESTAMP = 1
+
+class dds_presentation_access_scope(OrderedEnum):
+    DDS_PRESENTATION_INSTANCE = 0
+    DDS_PRESENTATION_TOPIC = 1
+    DDS_PRESENTATION_GROUP = 2
+
+class dds_ignorelocal(OrderedEnum):
+    DDS_IGNORELOCAL_NONE = 0
+    DDS_IGNORELOCAL_PARTICIPANT = 1
+    DDS_IGNORELOCAL_PROCESS = 2
+
+class dds_type_consistency(OrderedEnum):
+    DDS_TYPE_CONSISTENCY_DISALLOW_TYPE_COERCION = 0
+    DDS_TYPE_CONSISTENCY_ALLOW_TYPE_COERCION = 1
+
+class dds_qos_policy_id(Enum):
+    DDS_INVALID_QOS_POLICY_ID = 0
+    DDS_USERDATA_QOS_POLICY_ID = 1
+    DDS_DURABILITY_QOS_POLICY_ID = 2
+    DDS_PRESENTATION_QOS_POLICY_ID = 3
+    DDS_DEADLINE_QOS_POLICY_ID = 4
+    DDS_LATENCYBUDGET_QOS_POLICY_ID = 5
+    DDS_OWNERSHIP_QOS_POLICY_ID = 6
+    DDS_OWNERSHIPSTRENGTH_QOS_POLICY_ID = 7
+    DDS_LIVELINESS_QOS_POLICY_ID = 8
+    DDS_TIMEBASEDFILTER_QOS_POLICY_ID = 9
+    DDS_PARTITION_QOS_POLICY_ID = 10
+    DDS_RELIABILITY_QOS_POLICY_ID = 11
+    DDS_DESTINATIONORDER_QOS_POLICY_ID = 12
+    DDS_HISTORY_QOS_POLICY_ID = 13
+    DDS_RESOURCELIMITS_QOS_POLICY_ID = 14
+    DDS_ENTITYFACTORY_QOS_POLICY_ID = 15
+    DDS_WRITERDATALIFECYCLE_QOS_POLICY_ID = 16
+    DDS_READERDATALIFECYCLE_QOS_POLICY_ID = 17
+    DDS_TOPICDATA_QOS_POLICY_ID = 18
+    DDS_GROUPDATA_QOS_POLICY_ID = 19
+    DDS_TRANSPORTPRIORITY_QOS_POLICY_ID = 20
+    DDS_LIFESPAN_QOS_POLICY_ID = 21
+    DDS_DURABILITYSERVICE_QOS_POLICY_ID = 22
+    DDS_PROPERTY_QOS_POLICY_ID = 23
+    DDS_TYPE_CONSISTENCY_ENFORCEMENT_QOS_POLICY_ID = 24
+    DDS_DATA_REPRESENTATION_QOS_POLICY_ID = 25
+
+
+def to_kind_reliability(q: qos.Qos):
+    if qos.Policy.Reliability in q:
```

Comment:
Python really should have a shorthand notation for this pattern ... but I guess it is write once, check once, run forever ...

Reply:
yes i guess, i did not found a more pretty solution for that, i am open for ides. It will never change so should be ok